src/snowflake/cli/plugins/nativeapp/manager.py

```diff
@@ -254,6 +258,69 @@ def app_post_deploy_hooks(self) -> Optional[List[ApplicationPostDeployHook]]:
     def debug_mode(self) -> bool:
         return self.na_project.debug_mode
 
+    @contextmanager
+    def use_warehouse(self, new_wh: Optional[str]):
+        """
+        Switches to a different warehouse for a while, then switches back.
+        This is a no-op if the requested warehouse is already active.
+        If there is no default warehouse in the account, it will throw an error.
+        """
+        wh_result = self._execute_query(
+            f"select current_warehouse()", cursor_class=DictCursor
+        ).fetchone()
+        # If user has an assigned default warehouse, prev_wh will contain a value even if the warehouse is suspended.
+        prev_wh = wh_result["CURRENT_WAREHOUSE()"]
+
+        if new_wh is not None:
+            new_wh_id = to_identifier(new_wh)
```

Comment:
I think this `to_identifier` belongs in e.g. `nativeapp.project_model`, as that's where the YAML strings are turned into identifiers for other parts of the project definition.